platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/State.java

```diff
@@ -72,8 +75,35 @@ default void unregisterCommitListener(@NonNull final StateChangeListener listene
         throw new UnsupportedOperationException();
     }
 
+    /***
+     * {$inheritDoc}
+     */
     @Override
     default State copy() {
         throw new UnsupportedOperationException();
     }
+
+    /***
+     * {$inheritDoc}
+     */
+    @Nullable
+    @Override
+    default Hash getHash() {
+        throw new UnsupportedOperationException();
+    }
+
+    /***
+     * {$inheritDoc}
+     */
+    @Override
+    default void setHash(@NonNull Hash hash) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Hashes the state on demand if it is not already hashed. If the state is already hashed, this method is a no-op.
+     */
+    default void calculateHash() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Why have default methods if they just throw?
Would it not be better to have the compiler require implementing classes to provide these methods?

Reply:
They have this implementation to ease the burden on the test and helper classes that happened to implement `State` interface but don't need that functionality at all.